lib/ui/painting.dart

```diff
@@ -2600,6 +2588,48 @@ class ColorFilter {
   }
 }
 
+/// A [ColorFilter] that is backed by a native SkColorFilter.
+// We could not make [ColorFilter] be this, because it would no longer be const
+// constructible and would complicate the == and hashCode implementations.
```

Comment:
I would phrase this comment differently. The current comment is a commit message. Consider something like: "This is a private class, rather than being the implementation of the public ColorFilter, because we want ColorFilter to be const constructible and we want it to be efficiently comparable, so that widgets can check for ColorFilter equality to avoid repainting."

Reply:
Done